test/integration/options/file.spec.js

```diff
@@ -64,4 +66,82 @@ describe('--file', function () {
       done();
     });
   });
+
+  it('should run esm tests passed via file', function (done) {
+    const esmFile = 'collect-files.fixture.mjs';
+    const testArgs = ['--file', resolvePath(esmFile)];
+
+    runMochaJSON(esmFile, testArgs, function (err, res) {
+      if (err) {
+        return done(err);
+      }
+      expect(res, 'to have passed');
+      done();
+    });
+  });
+
+  it('should log a warning if a nonexistent file is specified', function (done) {
+    const nonexistentTestFileArg = 'nonexistent.test.ts';
+    runMocha(
+      nonexistentTestFileArg,
+      ['--file'],
+      function (err, res) {
+        if (err) {
+          return done(err);
+        }
+
+        expect(
+          res.output,
+          'to contain',
+          `Warning: Cannot find any files matching pattern`
+        ).and('to contain', nonexistentTestFileArg);
+        done();
+      },
+      {stdio: 'pipe'}
+    );
+  });
+
+  it('should provide warning for nonexistent cjs file extensions', function (done) {
+    const nonexistentCjsArg = 'nonexistent.test.js';
```

Comment:
[Naming] This confused me:

> nonexistent cjs file extensions
> `.test.js`

Is the intent that it's generally testing CommonJS files (`.js`)? I think the name could be a bit more clear. Maybe ...

```suggestion
  it('should provide warning for nonexistent js file extensions', function (done) {
    const nonexistentCjsArg = 'nonexistent.test.js';
```

Reply:
Yep just general testing of common JS files. Changed!